python/tools/polarfire_esb.py

```
@@ -47,64 +48,24 @@
-    elif args == "2507":
-        r = requests.get(url_2507)
-        open(fname_2507, "wb").write(r.content)
-        with zipfile.ZipFile(fname_2507) as zip_ref:
+
+def download_extract(current_file_cfg):
+    current_url = current_file_cfg.get("url", None)
+    filename = current_file_cfg.get("filename", None)
+    expected_md5 = current_file_cfg.get("md5", None)
+
+    if current_url:
+        r = requests.get(current_url)
```

Ruff (0.14.5)

Probable use of `requests` call without timeout

(S113)